cpp/src/tests/snmg_pagerank/snmg_pagerank_test.cu

```diff
@@ -0,0 +1,280 @@
+          //if (n_err > 0.001*m)
+          //{
+          //  eq(calculated_res,expected_res);
+          //}
+      }
+    }
+  }
+  template <typename idx_t,typename val_t>
+  void run_current_test(const MGPagerank_Usecase& param) {
+     const ::testing::TestInfo* const test_info =::testing::UnitTest::GetInstance()->current_test_info();
+     std::stringstream ss; 
+     std::string test_id = std::string(test_info->test_case_name()) + std::string(".") + std::string(test_info->name()) + std::string("_") + getFileName(param.matrix_file)+ std::string("_") + ss.str().c_str();
+
+     int m, k, nnz, n_gpus, max_iter=50;
+     val_t alpha = 0.85;
+     MM_typecode mc;
+
+     double t;
+
+     FILE* fpin = fopen(param.matrix_file.c_str(),"r");
+     
+     ASSERT_EQ(mm_properties<int>(fpin, 1, &mc, &m, &k, &nnz),0) << "could not read Matrix Market file properties"<< "\n";
+     ASSERT_TRUE(mm_is_matrix(mc));
+     ASSERT_TRUE(mm_is_coordinate(mc));
+     ASSERT_FALSE(mm_is_complex(mc));
+     ASSERT_FALSE(mm_is_skew(mc));
+     
+     // Allocate memory on host
+     std::vector<idx_t> cooRowInd(nnz), cooColInd(nnz), csrColInd(nnz), csrRowPtr(m+1);
+     std::vector<val_t> cooVal(nnz), csrVal(nnz), pagerank_h(m, 1.0/m), y_h(m, 0.0), y_ref(m, 0.0);
+
+     // Read
+     ASSERT_EQ( (mm_to_coo<int,val_t>(fpin, 1, nnz, &cooRowInd[0], &cooColInd[0], NULL, NULL)) , 0)<< "could not read matrix data"<< "\n";
+     ASSERT_EQ(fclose(fpin),0);
+     //ASSERT_EQ( (coo_to_csr<int,val_t> (m, m, nnz, &cooRowInd[0],  &cooColInd[0], NULL, NULL, &csrRowPtr[0], NULL, NULL, NULL)), 0) << "could not covert COO to CSR "<< "\n";
+     
+     // WARNING transpose happening here
+     coo2csr(cooColInd, cooRowInd, csrRowPtr, csrColInd);
+
+     CUDA_RT_CALL(cudaGetDeviceCount(&n_gpus));  
+     std::vector<size_t> v_loc(n_gpus), e_loc(n_gpus), part_offset(n_gpus+1);
+     random_vals(csrVal);
+     gdf_column *col_pagerank[n_gpus];
+
+     if (nnz<1200000000)
+     {
+       #pragma omp parallel num_threads(1)
+       {
+        //omp_set_num_threads(n_gpus);
+        auto i = omp_get_thread_num();
+        auto p = omp_get_num_threads(); 
+        CUDA_RT_CALL(cudaSetDevice(i));
+
+        #ifdef SNMG_VERBOSE 
+          #pragma omp master 
+          { 
+            std::cout << "Number of GPUs : "<< n_gpus <<std::endl;
+            std::cout << "Number of threads : "<< p <<std::endl;
+          }
+        #endif
+
+        gdf_column *col_off = new gdf_column, 
+                   *col_ind = new gdf_column, 
+                   *col_val = new gdf_column;
+        col_pagerank[i] = new gdf_column;
+        create_gdf_column(pagerank_h, col_pagerank[i]);
+        #pragma omp barrier
+
+        //load a chunck of the graph on each GPU 
+        load_csr_loc(csrRowPtr, csrColInd, csrVal, 
+                     v_loc, e_loc, part_offset,
+                     col_off, col_ind, col_val);
+        //printv(col_val->size,(float*)col_val->data,0);
```

Comment:
Remove comments

Reply:
Agreed but will be addressed in the next PR. This file is WIP and will be finalized along with the rest of the Pagernak solver. I'd rather do all the cleaning at once.